rider-fsharp/src/main/java/com/jetbrains/rider/plugins/fsharp/services/fsi/FsiHost.kt

```diff
@@ -1,22 +1,119 @@
+import com.jetbrains.rdclient.util.idea.LifetimedProjectComponent
 import com.jetbrains.rider.plugins.fsharp.*
+import com.jetbrains.rider.plugins.fsharp.services.fsi.consoleRunners.FsiConsoleRunnerBase
+import com.jetbrains.rider.plugins.fsharp.services.fsi.consoleRunners.FsiDefaultConsoleRunner
+import com.jetbrains.rider.plugins.fsharp.services.fsi.consoleRunners.FsiScriptProfileConsoleRunner
+import com.jetbrains.rider.plugins.fsharp.services.fsi.runScript.FSharpScriptConfiguration
+import com.jetbrains.rider.projectView.solution
+import com.jetbrains.rider.run.configurations.dotNetExe.DotNetExeConfigurationParameters
+import com.jetbrains.rider.run.createRunCommandLine
+import com.jetbrains.rider.runtime.DotNetExecutable
+import com.jetbrains.rider.runtime.DotNetRuntime
+import com.jetbrains.rider.runtime.RiderDotNetActiveRuntimeHost
+import com.jetbrains.rider.runtime.dotNetCore.DotNetCoreRuntimeType
+import com.jetbrains.rider.runtime.mono.MonoRuntime
+import com.jetbrains.rider.runtime.mono.MonoRuntimeType
+import com.jetbrains.rider.runtime.msNet.MsNetRuntimeType
 import org.jetbrains.concurrency.AsyncPromise
 import org.jetbrains.concurrency.Promise
+import org.jetbrains.concurrency.resolvedPromise
+import java.awt.event.FocusEvent
+import java.awt.event.FocusListener
+import java.util.*
+
+fun getFsiRunOptions(
+  project: Project,
+  configuration: FSharpScriptConfiguration? = null
+): Pair<DotNetExecutable, DotNetRuntime> {
+  val fsiHost = project.getComponent(FsiHost::class.java)
+  lateinit var sessionInfo: RdFsiSessionInfo
+  UIUtil.invokeLaterIfNeeded {
+    sessionInfo = fsiHost.rdFsiHost.requestNewFsiSessionInfo.sync(Unit)
+  }
+
+  val runtimeHost = RiderDotNetActiveRuntimeHost.getInstance(project)
+
+  val exePath = when (sessionInfo.runtime) {
+    RdFsiRuntime.Core -> runtimeHost.dotNetCoreRuntime.value!!.cliExePath
+    RdFsiRuntime.NetFramework -> sessionInfo.fsiPath
+    RdFsiRuntime.Mono -> {
+      val runtime = runtimeHost.getCurrentClassicNetRuntime(false).runtime as MonoRuntime
+      runtime.getMonoExe().path
+    }
+  }
+
+  val workingDirectory =
+    if (project.isDirectoryBased) {
+      val projectDir = project.guessProjectDir()
+      val workingDir =
+        if (projectDir != null && projectDir.exists()) projectDir
+        else VfsUtil.getUserHomeDir()
+      workingDir?.path ?: ""
+    } else ""
+
+
+  val runtimeType = when (sessionInfo.runtime) {
+    RdFsiRuntime.NetFramework -> MsNetRuntimeType
+    RdFsiRuntime.Mono -> MonoRuntimeType
+    RdFsiRuntime.Core -> DotNetCoreRuntimeType
+  }
+
+  val args = ArrayList<String>()
+  if (sessionInfo.runtime != RdFsiRuntime.NetFramework)
+    args.add(sessionInfo.fsiPath)
+  if (configuration != null)
+    args.add("--use:${configuration.scriptFile?.path}")
+  args.addAll(sessionInfo.args)
+
+  val parameters = DotNetExeConfigurationParameters(
+    project = project,
+    exePath = exePath,
+    programParameters = args.joinToString(" "),
```

Comment:
This will break if any of the args have spaces, so my recommendation is to use `ParameterListUtil` to join these.